### DISCUSSION TITLE
Comparison with LTeX

### DISCUSSION THREAD

**Discussion by CsatiZoltan** [davidlday/vscode-languagetool-linter#614] [Q&A] [answered]
Hi,

Could you please provide some advantages and disadvantages of this extension with respect to [LTeX](https://github.com/valentjn/vscode-ltex)?
Thank you.

**Comment by davidlday** [comment, marked as answer]
Hi @CsatiZoltan  - 

I'm not sure if I can answer that directly, but I can provide some idea of comparisons to help you decide what advantages and disadvantages there are. I have never used [LTeX](https://github.com/valentjn/vscode-ltex), so my answer is by nature very one-sided. 

1. I believe @valentjn is primarily focused on LaTeX whereas my primary focus is on Markdown.
2. He chose to embed LanguageTool in the extension whereas my goal is to reuse existing LanguageTool packages. That probably means the version of LanguageTool in his extension is explicitly tied to the version of the extension. It's a trade-off. Explicitly tying the versions together gives him more control over the backing service, but it also means more code. I personally don't have the time or resources to implement the same approach. I've looked through his very clean and professional code and considered doing something similar, but in the end I felt it was better to reuse and take a more modular approach. I'm in the process of separating out the code for managing LanguageTool services via Homebrew, Docker, Podman, custom install, and the public / premium provider so LT and be easily used in other TypeScript projects.  I already did something similar for the [Annotated Text](https://github.com/prosegrinder/annotatedtext) interface used by LT ([Markdown](https://github.com/prosegrinder/annotatedtext-remark), [HTML](https://github.com/prosegrinder/annotatedtext-rehype)) libraries used to send text to LT. They reuse existing parsers instead of implementing their own. In short, I don't want to be in the business of managing LanguageTool installs. I only want to do the minimum management of LT necessary to ensure it's up and running.
3. His implements the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/). While very nicely implemented, in my opinion it's unnecessary and maybe even inappropriate. LSP is designed for programming languages, not human languages. LT already implements a protocol designed for "linting" human language regardless of format, hence the Annotated Text libraries I mentioned above.
4. His community seems largely comprised of LaTeX users, which also likely means he has a contingent of skilled developers who are able / willing to contribute. As far as I can tell, my users are largely interested in writing prose. I've had contributors, for which I'm very thankful, but probably nowhere near as much participation as he has.
5. Both are the result of [LanguagTool Server](https://github.com/languagetool-language-server/languagetool-languageserver/) originally created by @adamvoss. Adam passed away in 2018. Shortly before passing away, he made me owner of his repositories. I didn't know Adam, and I don't know why he chose me. You can read more on this thread: https://github.com/languagetool-language-server/languagetool-languageserver/issues/6. @valentjn continued down the LSP path while I opted out of using LSP.

When it comes to picking one over the other, if you're working with LaTeX, use [LTeX](https://github.com/valentjn/vscode-ltex). I don't know if I'll ever implement a working parser for LaTeX. I've approached it a few times, but I know next to nothing about it, so I'm not convinced I could do it any kind of justice. Beyond that, his is also better maintained. Between my day job, my family, and my writing, I haven't had a lot of time to devote to this extension. I have no intention of abandoning it, but the long list of issues and PRs speaks for itself.

Hope this helps. And please ask questions or even provide your own perspective if you've had a chance to try both extensions out. In the spirit of open source, this should be a dialogue, not an argument. 😄 

@valentjn - I apologize if I used the wrong pronouns for you. Happy to correct them if needed. Would also love for you to share your thoughts as well, and please correct anything I may have gotten wrong.

**Reply by CsatiZoltan** [reply]
Thank you for taking your time to write such a detailed and insightful reply. I use both Markdown and LaTeX, so I installed both extensions. I will come back to you with my impressions.

**Comment by CsatiZoltan** [comment]
@davidlday For info, LTeX is now [archived](https://github.com/valentjn/vscode-ltex), but there is [LTeX+](https://github.com/ltex-plus/vscode-ltex-plus). It has the option to [connect to a self-hosted version](https://ltex-plus.github.io/ltex-plus/settings.html#ltexlanguagetoolhttpserveruri) of LanguageTool if you do not want to use the bundled version. I haven't tried it yet.
